File: Project 1 Question 4/src/Zookeeper.java
import java.util.*;

/**
 * The Zookeeper class is a helper which create object for all the animals 
 * and run different tasks on them.
 
 */

public class Zookeeper {

	/**
	 * Arraylist to contain all the animal object
	 */
	protected ArrayList<Animal> al;
	
	
	/**
	 * Constructor to initiate class member
	 */
	public Zookeeper()
	{
		al = new ArrayList<Animal>();
	}
	
	
	/**
	 * This method create all the animal object based on first character
	 * of the animal name and add them to the arraylist
	 * @param z  Arraylist of animals'names
	 */
	public void addAnimals(ArrayList<String> z)
	{
		
		for(int i=0;i<z.size();i++)
		{
			switch(z.get(i).charAt(0))
			{
			case 'C':al.add(new Cat(z.get(i)));
					break;
			case 'D':al.add(new Dog(z.get(i)));
					break;
			case 'E':al.add(new Elephant(z.get(i)));
					break;
			case 'L':al.add(new Lion(z.get(i)));
					break;
			case 'H':al.add(new Hippo(z.get(i)));
					break;
			case 'W':al.add(new Wolf(z.get(i)));
					break;
			case 'T':al.add(new Tiger(z.get(i)));
					break;
			case 'R':al.add(new Rhino(z.get(i)));
					break;
			
	
			}
		}
	}
	/**
	 * Method to wake up the animal
	 */
	public void wakeup()
	{
		System.out.println("WAKING UP THE ANIMALS");
		for(int i=0;i<al.size();i++)
		{
			al.get(i).wakeUp();
		}
		
	}
	
	/**
	 * Method to feed the animal
	*/
	public void feed()
	{
		System.out.println("FEEDING THE ANIMALS");
		for(int i=0;i<al.size();i++)
		{
			al.get(i).eat();
		}
		
	}
	
	/**
	 * Method which tell animal to exercise
	 */
	public void exercise()
	{
		System.out.println("EXERCISING THE ANIMALS");
		for(int i=0;i<al.size();i++)
		{
			al.get(i).roam();
		}
		
	}
	
	/**
	 * Method to get the animal make noise
	 */
	public void rollCall()
	{
		
		System.out.println("ROLL CALL THE ANIMALS");
		for(int i=0;i<al.size();i++)
		{
			al.get(i).makeNoise();
		}
	}
	
	/**
	 * Method which ask animal to sleep
	 */
	public void shutdown()
	{
		
		System.out.println("SHUTTING THE ZOO");
		for(int i=0;i<al.size();i++)
		{
			al.get(i).sleep();
		}
	}
	/**
	 * Method to execute the tasks
	 */
	public void carryOutTask()
	{
		this.wakeup();
		System.out.println("");
		this.rollCall();
		System.out.println("");
		this.feed();
		System.out.println("");
		this.exercise();
		System.out.println("");
		this.shutdown();
	}
}

File: Project 2/Zookeeper.cpp
#include<iostream>
using namespace std;

#include "Zookeeper.h"
#include "Feline.h"
#include "Canine.h"
#include "Pachyderm.h"

Zookeeper::Zookeeper(vector<string> animals)
{
    for(int i=0;i<animals.size();i++)
    {
      switch(animals[i][0])
      {
        case 'C':animal_list_.push_back(new Cat(animals[i]));
                break;
        case 'D':animal_list_.push_back(new Dog(animals[i]));
                break;
        case 'E':animal_list_.push_back(new Elephant(animals[i]));
                break;
        case 'T':animal_list_.push_back(new Tiger(animals[i]));
                break;
        case 'W':animal_list_.push_back(new Wolf(animals[i]));
                break;
        case 'L':animal_list_.push_back(new Lion(animals[i]));
                break;
        case 'H':animal_list_.push_back(new Cat(animals[i]));
                break;
        case 'R':animal_list_.push_back(new Rhino(animals[i]));
                break;

      }
    }
}
void Zookeeper::Feed()
{
cout<<"FEEDING THE ANIMALS"<<endl;

for(int i=0;i<animal_list_.size();i++)
{
  animal_list_[i]->StartEat();
}
}
void Zookeeper::RollCall()
{
  cout<<"ROLL CALL THE ANIMALS"<<endl;

  for(int i=0;i<animal_list_.size();i++)
  {
    animal_list_[i]->Roam();
  }
}




void Zookeeper::WakeUp()
{
  cout<<"WAKING UP THE ANIMALS"<<endl;

  for(int i=0;i<animal_list_.size();i++)
  {
    animal_list_[i]->WakeUp();
  }
}

void Zookeeper::ShutDown()
{
  cout<<"SHUTTING THE ZOO"<<endl;

  for(int i=0;i<animal_list_.size();i++)
  {
    animal_list_[i]->Sleep();
  }
}

void Zookeeper::Exercise()
{
  cout<<"ROLL CALL THE ANIMALS"<<endl;

  for(int i=0;i<animal_list_.size();i++)
  {
    animal_list_[i]->Roam();
  }
}

void Zookeeper::CarryOutTask()
{
  
    WakeUp();
    cout<<endl;

    RollCall();
    cout<<endl;

    Feed();
    cout<<endl;

    Exercise();
    cout<<endl;
    ShutDown();
    cout<<endl;

  
}